### ISSUE TITLE
Country Not Shown as a List

### ISSUE THREAD

**Issue by Andy9999** [dnnsoftware/Dnn.Platform#4624]
1. After a fresh install of 9.9.0 or 9.9.1 the country field's list is not shown. The lists entries still exist in the Lists table. However country is just a text field and not a combo box. Flagging the field as required has no effect. Flagging both country and region as required leads to the country text control showing "N/A" and offering a text input for region (no combo box) :(

![dnn country](https://user-images.githubusercontent.com/547819/116080050-e0b30300-a698-11eb-95b3-248ef9981209.png)

![country region mandatory](https://user-images.githubusercontent.com/547819/116080314-312a6080-a699-11eb-85ee-791152906394.png)

![first login](https://user-images.githubusercontent.com/547819/116080576-89f9f900-a699-11eb-92de-3da86e7073d2.png)

2. For some strange reason, if upgrading an existing site to 9.9.0/9.9.1 the list of countries is shown. Looking at a profile shows it properly. But there is a 2nd problem. If configuring a custom login page and making country and region mandatory a user gets forced to enter country and region during the first login. In that configuration the custom login page shows both region fields, the combo box and the text field. The text field is not in its proper position for some reason. After selecting a country the region combo box is properly filled with the regions if the country has regions. At all times both region fields are shown (text and combo). After selecting a region and saving the profile the region combo box resets to "Not specified" and the profile is not saved so the user is stuck at that point and can never login. If there is no custom login page configured it works (but cannot be styled). Also it works if editing the profile the regular way.

I filed both in one bug report because it seems to be related. On our production site (which was upgraded to 9.9.0) we have only country mandatory. When upgrading to 9.4.1 a while ago, I had to hide the region text field (via CSS) and I made region not required so that users can register and login successfully. However, the selected region during the enforced profile update whn logging in for the first time is not stored and has to be updated again by visiting the profile. So long story short, this bug seems to be around for a while.

If someone has any idea what I should look at, please let me know and I'll try some debugging if no one else has time.

**Comment by Andy9999** [comment]
I just realized that there is no more combo boxes showing LOL. And a default installation only has US regions installed. So the bug 1) doesn't exist. 

The updated site shows a combo box for some reason but is not working properly as described above (2.).

**Comment by Andy9999** [comment]
OK.....I could successfully reproduce bug 2) on a fresh 9.9.1 install.

Steps to reproduce:
1. Make country and region mandatory.
2. Add a custom login page (with proper permissions for unauthenticated users and not showing in menu)
3. Register a new user
4. Authorize the new user
5. Login as the new user for the first time:
![first login](https://user-images.githubusercontent.com/547819/116087084-ca10aa00-a6a0-11eb-81c6-789d64632486.png)

**Comment by sleupold** [comment]
AFAIR both comboboxes had been replaced in ~DNN 9.5.0 by text boxes with type suggest. If no countries and regions are listed in your new install, please check browser console for script errors, tank you.

**Comment by Andy9999** [comment]
I added a simple list to reproduce the issue above.

**Comment by Andy9999** [comment]
One more update......if following step 1-5 you can see that the type suggest isn't working either under those conditions. So the user is actual stuck at that point because the region is mandatory. Even if entering something into the misplaced region text field you cannot continue:
![stuck](https://user-images.githubusercontent.com/547819/116090457-2e813880-a6a4-11eb-9bd0-31dce5d9a048.png)

**Comment by sleupold** [comment]
I just tested with DNN 9.9.0 (and there hasn't been any changes in this area for DNN 9.9.1 AFAIK), and type suggest worked for Country (though I had ti work around the auto fill of my browser) and after filling in the country, a dropdown for Region appeared.

**Comment by Andy9999** [comment]
Did you follow all the steps above and used a custom login form?

**Comment by sleupold** [comment]
yes, I tested with default registration and custom registration form, using default profile definition.
You shouldn't try to override the lists for Country and Region, which are delivered with DNN.

**Comment by Andy9999** [comment]
I only did step 1 through 5. No overriding of any lists. Default install, so there is only countries and regions for US.

**Comment by sleupold** [comment]
sorry, I tried with custom registration form, not custom login - how did you implement it?

**Comment by Andy9999** [comment]
I just made a new page, dropped the "Account login" module in it. Changed the page to not show up in the menu (don't think that matters) and set page permissions for "All users" to show.

**Comment by sleupold** [comment]
Did you require country and region properties for registration or just for login?

Von: Andy9999 ***@***.***>
Gesendet: Montag, 26. April 2021 15:40
An: dnnsoftware/Dnn.Platform ***@***.***>
Cc: Sebastian Leupold ***@***.***>; Comment ***@***.***>
Betreff: Re: [dnnsoftware/Dnn.Platform] Country Not Shown as a List (#4624)


I just made a new page, dropped the "Account login" module in it. Changed the page to not show up in the menu (don't think that matters) and set page permissions for "All users" to show.

**Comment by Andy9999** [comment]
For registration. I changed it in Site Settings -> Site behavior -> User Profiles and made both required:
![country region mandatory](https://user-images.githubusercontent.com/547819/116092943-891b9400-a6a6-11eb-8aca-27fdc8072049.png)

**Comment by sleupold** [comment]
Does the profile page behave as expected, when editing a user as admin?

**Comment by Andy9999** [comment]
Editing the profile always works. It only occurs during the first login when the user is forced to update the profile.

**Comment by Andy9999** [comment]
Most likely the login control could be styled as well, but using a custom login page is a lot better and additionally supports multi-language. We have for example some additional text above the form (HTML module) giving further explanations.

**Comment by sleupold** [comment]
I just tried with default login page and had no issues filling in both missing properties.

**Comment by Andy9999** [comment]
Yes, default login page works but no custom login page.

**Comment by sleupold** [comment]
I just verified, on a custom login page Country is not displayed as Text Suggest and Region dropdown doesn't get populated.

**Comment by Andy9999** [comment]
OK, great :)  Thank you.

Would be great if this issue could get fixed or if someone could point me in the right direction to fix it. At first I thought it was probably something in our sites configuration and therefore I tried it on a fresh install.

**Comment by sleupold** [comment]
Most likely, code in Account Login module did not get improved like in default login control, wenn @donker implemented text suggest for Country in ~DNN 9.5.0

**Comment by Andy9999** [comment]
But this problem was there in 9.4.1 as well. Back then I introduced some CSS to hide that out-of-position (region) text box you can see in my screenshot above.

**Comment by sleupold** [comment]
My memor might be wrong about the version, the feature was added.

**Comment by Andy9999** [comment]
OK......9.4.1 is close to 9.5.0 :)

**Comment by Andy9999** [comment]
Same problem. Looks exactly the same as 9.9.0/9.9.1. 9.4.1 had text suggest as well for country.

**Comment by Andy9999** [comment]
And actually the new suggestion text control is good for mobile, no criticism about that. It just confused me during testing. Don't know if there is a possibility to let users know it's not just a simple text field. For mobile this is probably standard and quite convenient and easier than clicking on a small button (touch) of a combo box control.

For now I have given it a lower priority on my to-do list and will visit it later when more important things have been completed. Maybe the script sent down to the browser has some problem that only occurs on custom login pages and only with the country/region field.

**Comment by pollardo84** [comment]
Hi!

I confirm the bug is still there, exactly as described on a fresh installation of DNN 9.11.2.

@Andy9999, were you ever able to fix this somehow?

Thank you.

**Comment by pollardo84** [comment]
@donker thanks for looking into this.

Please follow the steps described by @Andy9999 and you will see the issue. Here's a summary:

You need to create a custom login page, drop a login module and make it visible to everyone.
Also make the country a mandatory field for users.
Then create a new user from the admin panel and enter only the basic stuff (name, email, password) and leave the rest blank for the client to complete on the first login.
On different browser or an incognito window, try loging in with that user you created. (on the custom login page). The form will appear showing the missing details for you to complete. There you will see the issue with the country field.

I hope I was clear and you are able to replicate this.

Thank you!

**Comment by rodrigoratan** [comment]
i will try to look into this tomorrow at https://discord.gg/rfgckvdY?event=1101633626157875300

**Comment by donker** [comment]
I can confirm the issue. Drilling down the necessary dnn.CountriesRegions.js and dnnSF are not loaded by the client. It looks like it's an issue with the fact that the registration form loads in an iFrame in the mentioned scenario.

**Comment by Timo-Breumelhof** [comment]
I just ran into this too on 09.13.10
